message_converter/src/converter_node.cpp

```diff
@@ -98,20 +37,23 @@ void SwiftNavRover::baselinePositionCallback(const libsbp_ros_msgs::MsgBaselineN
     int tow = msg->tow;
     int h_accuracy = msg->h_accuracy;
     int v_accuracy = msg->v_accuracy;
-    boost::array<double, 9> covariance = {{h_accuracy, 0, 0, 0, h_accuracy, 0, 0, 0, v_accuracy}};
+    boost::array<double, 9> covariance = {{h_accuracy/1000.00, 0, 0, 0, h_accuracy/1000.00, 0, 0, 0, v_accuracy/1000.00}};
     int n_sats = msg->n_sats;
     int fixed_mode = (msg->flags) & (0b00000111);
+    SwiftNavRover::positioning_mode_ = fixed_mode;
     ros::Time t = ros::Time::now();
     if(n_sats == 0){
-        SwiftNavRover::satelliteCheck = false;
+        SwiftNavRover::satellite_check_ = false;
+        SwiftNavRover::publishRTKAvailable(t);
```

Comment:
instead of using `now()`; preserve the timestamp of the incoming `MsgBaselineNed` message.

Reply:
The messages are stamped with TOW from the origin, which is the reason why `now()` has been used here.

Reply:
They do [have GPS TOW](https://github.com/ethz-asl/ethz_piksi_ros/blob/master/piksi_rtk_msgs/msg/BaselineNed.msg), but is `header.stamp` not populated?

Reply:
We are not using the message definition you are referring to. The messages we use are from [libsbp_ros_msgs](https://github.com/ethz-asl/ethz_piksi_ros/tree/master/libsbp_ros_msgs). You wont find the message definitions in this repository though, since they use a script to auto-generate the messages. 

Here is the message `MsgBaselineNED`
```

# This message is automatically generated using generator.py
# PLEASE DO NOT MODIFY MANUALLY.

# Navigation
# MSG_BASELINE_NED 0x020C
#
# Baseline in NED
#
# This message reports the baseline solution in North East Down
# (NED) coordinates. This baseline is the relative vector distance
# from the base station to the rover receiver, and NED coordinate
# system is defined at the local WGS84 tangent plane centered at the
# base station position.  The full GPS time is given by the
# preceding MSG_GPS_TIME with the matching time-of-week (tow).

# GPS Time of Week
uint32 tow # [ms]

# Baseline North coordinate
int32 n # [mm]

# Baseline East coordinate
int32 e # [mm]

# Baseline Down coordinate
int32 d # [mm]

# Horizontal position estimated standard deviation
uint16 h_accuracy # [mm]

# Vertical position estimated standard deviation
uint16 v_accuracy # [mm]

# Number of satellites used in solution
uint8 n_sats # []

# Status flags
uint8 flags # []
```

Reply:
marking this as resolved